protocols/identify/src/handler.rs

```diff
@@ -152,19 +161,24 @@ impl Handler {
         >,
     ) {
         match output {
-            future::Either::Left(substream) => {
+            future::Either::Left(stream) => {
                 let info = self.build_info();
 
-                self.pending_replies
-                    .push(crate::protocol::send(substream, info).boxed());
+                if self
+                    .identify_futs
+                    .try_push(crate::protocol::identify_send(stream, info))
+                    .is_err()
+                {
+                    warn!("Dropping inbound stream because we are at capacity");
+                }
             }
-            future::Either::Right(fut) => {
-                if self.inbound_identify_push.replace(fut).is_some() {
-                    warn!(
-                        "New inbound identify push stream from {} while still \
-                         upgrading previous one. Replacing previous with new.",
-                        self.remote_peer_id,
-                    );
+            future::Either::Right(stream) => {
+                if self
+                    .identify_futs
+                    .try_push(recv_push(stream).boxed())
```

Comment:
Let's be consistent in how we refer to these functions. Because we are declaring the `protocol` module further up, I think we don't need the `crate::`.

```suggestion
                    .try_push(protocol::recv_push(stream))
```